### PR TITLE
[enhancement](page cache) use separate pk index cache

### PR DESCRIPTION
**Pull request by yixiutt**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by yixiutt** [comment]
run buildall

**Comment by yixiutt** [comment]
run buildall

**Review by xinyiZzz — DISMISSED** [review]
LGTM

File: be/src/olap/rowset/segment_v2/indexed_column_reader.cpp [hunk 1]
```diff
@@ -98,6 +98,9 @@ Status IndexedColumnReader::read_page(const PagePointer& pp, PageHandle* handle,
     opts.use_page_cache = _use_page_cache;
     opts.kept_in_memory = _kept_in_memory;
     opts.type = type;
+    if (_is_pk_index) {
+        type = PRIMARY_KEY_INDEX_PAGE;
```
Comment (zhannngchen) [inline be/src/olap/rowset/segment_v2/indexed_column_reader.cpp:102]: should be `opts.type = PRIMARY_KEY_INDEX_PAGE`?

**Review by zhannngchen — CHANGES_REQUESTED** [review]
pls address the comment

**Comment by yixiutt** [comment]
run buildall

**Review by zhannngchen — APPROVED** [review]
LGTM

**Pull request merged**